### PR TITLE
Another take on bug #4953

### PR DESCRIPTION
**Pull request by oversizedhat**
## Proposed changes

Bug fix to https://github.com/webdriverio/webdriverio/issues/4953, as discussed in https://github.com/webdriverio/webdriverio/pull/5040

Hard to say if this solves all issues. I have not added any kind of extensive testing of it but it at least looks like it solves our case where we use mocha framework in combo with wdio. Have not tested any other scenarios like when using cucumber.

The "currentTest" addition to the hook seems to have been expected as its is already used (but ignored) in https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-mocha-framework/src/index.js

## Types of changes

- [x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-allure-reporter/src/index.js [hunk 1]
```diff
@@ -100,11 +100,20 @@ class AllureReporter extends WDIOReporter {
     }
 
     onTestStart(test) {
+        const testTitle = test.currentTest ? test.currentTest : test.title
+        if (this.isAnyTestRunning()) {
+            if (this.allure.getCurrentTest().name == testTitle) {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/index.js:105]:
```suggestion
        if (this.isAnyTestRunning() && this.allure.getCurrentTest().name == testTitle) {
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I think if this solves the issue for at least Mocha test it is already a step forward. One comment on the implementation.

@BorisOsipov any thoughts?

**Comment by BorisOsipov** [comment]
@christian-bromann I dont mind.

**Comment by christian-bromann** [comment]
@oversizedhat please address the comment above and add a unit test for your changes in the Allure reporter. Let me know if you need and help.

**Comment by oversizedhat** [comment]
cheers @christian-bromann I will take a look into adding some tests

**Comment by christian-bromann** [comment]
@oversizedhat the change looks good to me, can you squash all commits into one so that the CLA bot is happy? Cheers!

**Comment by oversizedhat** [comment]
cheers @christian-bromann, will try to wrap it up. Still want to add some kind of test to validate the change too, it was a valid comment.

**Comment by oversizedhat** [comment]
@christian-bromann cleaned it up now and added at least some basic test to validate that there are no test duplicates in the allure report due to hook start. There is room for more testing but its something at least

**Review by christian-bromann — APPROVED** [review]
Awesome, LGTM 👍

**Pull request merged**